inception-project/inception
# Feature/2186 support per project guest annotators

**What's in the PR**
- Add "realm" concept to user accounts
- If multiple "realms" are in use, then allow switching between them in the user management page
- Allow force-logging out users via the SessionRegistry when users are deleted by registering a ConcurrentSessionFilter
- Extended invitation acceptance page to allow login by nick creating a project-local user
- Delete project-local users when the project is deleted
- Set a realm on the user object when logging in via preauthentication
- Added new variations of the visibleWhen LambdaBehavior
- Simplify NameUtil helper class
- Avoid browser pre-filling fields in the user management page
- Display UI name of user in the menu bar if possible
- Respect login limit on invite page
- Allow configuring the placeholder for the user ID when using passwordless invites
- Improve displaying of username on the project users panel
- Allow adding an invitation text to invites
- Hard-wire the remote API to using the internal DB for authentication even if external authentication is enabled
- Use the InceptionDaoAuthenticationProvider to avoid that people without a password log in via "normal" mechanisms
- Importing and exporting of project invite settings
- Importing and exporting of project-bound users
- When importing users, set an empty password and add the USER role to avoid potential NPEs when using these users via pre-authentication
- Added unit tests
- Updated documentation

**How to test manually**
* Sign in normally
* Sign in via invite with normal account
* Sign in via invite using a nick
* Delete project and see that nick accounts get automatically removed
* Check switching between realms on user management page
* Check usage of display name in different parts of the UI

**Automatic testing**

**Documentation**
